File: backend/HSE_Football_Backend/Models/PlayerApplication.cs
using System.ComponentModel.DataAnnotations;

namespace HSE_Football_Backend.Models
{
	/// <summary>
	/// Заявка игрока
	/// </summary>
	public class PlayerApplication
	{
		/// <summary>
		/// ID заявки игрока
		/// </summary>
		[Key]
		public long Id { get; set; }

		/// <summary>
		/// номер телефона игрока
		/// </summary>
		public string? PlayerPhoneNumber { get; set; }

		/// <summary>
		/// Игровое амплуа
		/// Ниже представлены степени двойки, то есть значение для ЦФРВ - 2^0 = 1
		/// 0 - ЦФРВ
		/// 1 - ЛФРВ
		/// 2 - ПФРВ
		/// 3 - ЦАП
		/// 4 - ЦОП
		/// 5 - ЛП
		/// 6 - ПП
		/// 7 - ЦЗ
		/// 8 - ЛЗ
		/// 9 - ПЗ
		/// 10 - ВРТ
		/// </summary>
		[Range(0, 2047, ErrorMessage = "Некорректное амплуа")]
		public int FootballPosition { get; set; }

		/// <summary>
		/// Предпочтительные турниры
		/// Ниже представлены степени двойки, то есть значение для Второй и Первой лиг - 2^6 + 2^7 = 192
		/// 0 - Осенний кубок
		/// 1 - Футзал Молодежная лига
		/// 2 - Футзал Вторая лига
		/// 3 - Футзал Первая лига
		/// 4 - Футзал Высшая лига
		/// 5 - Молодежная лига
		/// 6 - Вторая лига
		/// 7 - Первая лига
		/// 8 - Высшая лига
		/// 9 - Летний кубок
		/// </summary>
		[Range(0, 1023, ErrorMessage = "Некорректные турниры")]
		public int PreferredTournaments { get; set; }

        /// <summary>
        /// Факультет, курс (для выпускников - год выпуска)
        /// </summary>
        [StringLength(101, ErrorMessage = "Слишком много символов")]
        public string? Faculty { get; set; }

        /// <summary>
        /// Футбольный опыт
        /// </summary>
        [StringLength(201, ErrorMessage = "Слишком много символов")]
        public string? FootballExperience { get; set; }

        /// <summary>
        /// Опыт в турнирах ВШЭ
        /// </summary>
        [StringLength(201, ErrorMessage = "Слишком много символов")]
        public string? TournamentExperience { get; set; }

        /// <summary>
        /// Контактная информация
        /// </summary>
        [StringLength(41, MinimumLength = 2, ErrorMessage = "Контактная информация должна быть от 2 до 40 символов")]
        public string? Contact { get; set; }

        /// <summary>
        /// ФИО
        /// </summary>
        [StringLength(61, MinimumLength = 3, ErrorMessage = "ФИО должно быть от 3 до 60 символов")]
        [RegularExpression(@"[ А-Яа-яЁёA-Za-z]+$", ErrorMessage = "В ФИО могут присутствовать только буквы и пробелы")]
        public string? Name { get; set; }

        /// <summary>
        /// Фото профиля
        /// </summary>
        public string? Photo { get; set; }

        /// <summary>
        /// Показывает возможное наличие нецензурной лексики
        /// </summary>
        public bool Attention { get; set; }
    }
}

File: android/sources/features/profile/domain_profile/src/main/kotlin/andryuh/football/domain_profile/dto/PlayerApplication.kt
package andryuh.football.domain_profile.dto

import android.os.Parcelable
import androidx.compose.runtime.Immutable
import andryuh.football.core_models.PlayerPosition
import andryuh.football.core_models.Tournament
import andryuh.football.core_network.serializers.EnumListAsIntSerializer
import kotlinx.parcelize.Parcelize
import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Immutable
@Parcelize
@Serializable
data class PlayerApplication(
  val id: String,
  @SerialName("playerPhoneNumber") val phoneNumber: String,
  @Serializable(with = PlayerPositionsSerializer::class) val footballPosition: List<PlayerPosition>,
  @Serializable(with = TournamentsSerializer::class) val preferredTournaments: List<Tournament>,
  val footballExperience: String?,
  @SerialName("tournamentExperience") val tournamentsExperience: String?,
  val contact: String?,
  val name: String?,
  @SerialName("photo") val imageUrl: String?
) : Parcelable

internal class PlayerPositionsSerializer :
  EnumListAsIntSerializer<PlayerPosition>(
    serialName = "footballPosition",
    enumArray = PlayerPosition.values(),
  )

internal class TournamentsSerializer :
  EnumListAsIntSerializer<Tournament>(
    serialName = "preferredTournaments",
    enumArray = Tournament.values(),
  )
